Paralayout/Angle.swift

```diff
@@ -0,0 +1,146 @@
+//
+//  Copyright © 2020 Square, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import CoreGraphics
+
+public struct Angle: Comparable {
+
+    // MARK: - Public Static Properties
+
+    /// A zero angle.
+    public static let zero = Angle(radians: 0)
+
+    /// A right angle, representing 1/4 of a circle (90º, or π/2 radians).
+    public static let right = Angle(radians: .pi / 2)
+
+    /// A straight angle, representing 1/2 of a circle (180º, or π radians).
+    public static let halfCircle = Angle(radians: .pi)
+
+    /// An angle representing a full rotation around a circle (360º, or 2π radians).
+    public static let fullCircle = Angle(radians: 2 * .pi)
+
+    // MARK: - Life Cycle
+
+    /// Create an angle in radians.
+    public init(radians: CGFloat) {
+        self.radians = radians
+    }
+
+    /// Create an angle in degrees.
+    public init(degrees: CGFloat) {
+        self.radians = Angle.radians(fromDegrees: degrees)
+    }
+
+    /// Create an angle, measured as the angle between two points, in the range `(-180°,180°]`.
```

Comment:
Is this intentionally a left-paren and a right-bracket to indicate exclusive/inclusive range? Or is that a typo?

Reply:
Yep, intentional exclusive/inclusive range.